### PR TITLE
Chatapp

### PR DESCRIPTION
**Pull request by Dmarceli**
containerized backend and frontend, started with methods and channels selections 

Issues:
- #5 
- #4

### CODE REVIEW

**Review by PukieDiederik — CHANGES_REQUESTED** [review]
Hello,
As promised I will give an explanation of why I can't accept this pull request.

First of all, let's talk about the `.env` file. You removed this from the gitignore and added it to git. A `.env` file should **NEVER** be added to git. A `.env` (otherwise known as the environment file) is used to configure the project for YOUR enviroment. Your environment might look different from mine. In this file you also usually add sensitive information (like here you added the DB_PASS which is sensitive information) which you NEVER WANT ON YOUR GIT. Removing it and commiting doesn't solve the issue (git is a list of changes, so removing your env file means its still visible in older commits). The `.env.example` file has also been deleted. This file serves to show which variables you need to/can set. Now if no .env file is present people wont easily understand how to configure our project.

Another thing, commiting to branches after creating a pull request. When you commit to your branch after having made a pull request will invalidate any review made on that branch (it is configured this way). Normally you don't need to push to the branch after a PR unless changes are requested/you needed to fix something. If you want to keep working on that feature it probably means it was not completed. If you want to continue working with those changes implemented but on another feature, you create a new branch.

What should be in a pull request? One feature. In this case it has added: the backend initial setup, the frontend initial setup, a chat app, database generation, endpoints. All of these should be seperate PR's. Now it is really hard to check everything you have done (because now I have to look through 11k+ changes instead of having the init be ~9k, and then having 3-4 other pull requests with only a couple hundred changes).

Last, the pull request message. A pull request message should give you a good overview of what has changed, why it has changed and what issues are related to a PR. It also helps if you give resources on how to test your code.

How I would like to see this PR split. First we should have 2 PRs for the initial setup of the frontend and the backend. Both of them should include a `dockerfile` and modify the `docker-compose` accordingly. After this we should have 3 seperate PRs for the database, the endpoints, and the chat app. (I don't think I need to say this but these should all be on seperate branches.) I recommend going back to the master branch (which should NOT have been modified yet) and creating new branches from there. Then you can copy over the files from this branch when you need them.

File: docker-compose.yml [hunk 1]
```diff
@@ -8,8 +8,39 @@ services:
...
+    working_dir: /home/node/app
+    user: node
+    entrypoint: ["bash" , "-c" , "npm install && npm run start:dev"]
+  frontend:
+    image: node
+    restart: on-failure
+    volumes:
+      - ./frontend:/home/node/app
+    ports:
+      - '5173:5173'
```
Comment (PukieDiederik) [inline docker-compose.yml:37]: Might be worth changing to port 80?

File: backend/src/users/users.controller.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { Controller, Get, Post, Body, Patch, Param, Delete, Res } from '@nestjs/common';
+import { UsersService } from './users.service';
+
+
+@Controller('users')
+export class UsersController {
+  constructor(private readonly usersService: UsersService) {}
+
```
Comment (PukieDiederik) [inline backend/src/users/users.controller.ts:8]: [REFACTORING] The endpoints

File: backend/src/chat/chat.entity.ts
Comment (PukieDiederik) [inline backend/src/chat/chat.entity.ts:1]: Rename this module to "message"

File: backend/src/chat/chat.entity.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    CreateDateColumn,
+   } from 'typeorm';
+    
+   @Entity()
+   export class Chat {
+    @PrimaryGeneratedColumn()
+    id: number;
+    
+    @Column()
+    author: string;
```
Comment (PukieDiederik) [inline backend/src/chat/chat.entity.ts:14]: Should reference a user instead of a string

File: backend/src/chat/chat.entity.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    
+    @Column()
+    message: string;
+    
+    @CreateDateColumn()
+    time: Date;
+
+    @Column()
+    channel: number;
```
Comment (PukieDiederik) [inline backend/src/chat/chat.entity.ts:23]: should reference a channel

File: backend/src/game_history/game_history.entity.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    user_id_loser: User;
+
+    @Column()
+    time_begin: Date;
+    
+    @Column()
+    time_end: Date;
+
+    @Column()
+    points: number;
```
Comment (PukieDiederik) [inline backend/src/game_history/game_history.entity.ts:31]: [REFACTORING] rename to `xp`, `xp_gained` or something similar

File: frontend/src/App.vue [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+    </div>
+  </div>
+</template>
+
+<script setup>
+import { io } from 'socket.io-client'
+import { ref, onBeforeMount, watch, nextTick } from 'vue';
+
+
+const socket = io('http://localhost:3000');
```
Comment (PukieDiederik) [inline frontend/src/App.vue:48]: recommend not hard coding this

File: docker-compose.yml [hunk 7]
```diff
@@ -8,8 +8,39 @@ services:
...
+    container_name: frontend
+    working_dir: /home/node/app
+    depends_on:
+      - backend
+    user: node  
+    entrypoint: ["bash" , "-c" , "npm install && npm run dev"]
```
[the remark this answers is not in the document]
Comment (PukieDiederik) [reply docker-compose.yml:43]: Requested changes:

- [ ] Update makefile.
- [ ] Create docker files.
- [ ] rename chat table to something like `message`
- [ ] On "chat" do not use numbers or strings and instead use the correct entity
- [ ] Move database modules to seperate folder.

**Review by PukieDiederik — COMMENTED** [review]
.

File: frontend/src/App.vue (hunk 6 above)
Comment (ncameiri) [reply frontend/src/App.vue:48]: To be adressed on issue #9

**Pull request closed without merging**